internal/dag/dag.go

```diff
@@ -391,29 +403,69 @@ func (d *DAG) recompute() dag {
 		_d.roots = append(_d.roots, svh)
 	}
 
-	return _d
+	for meta, orph := range orphaned {
+		if orph {
+			ir, ok := d.ingressroutes[meta]
+			if ok {
+				status = append(status, ingressrouteStatus{object: ir, status: "orphaned", msg: "this IngressRoute is not part of a delegation chain from a root IngressRoute"})
+			}
+		}
+	}
+	return _d, status
 }
 
-func (d *DAG) processIngressRoute(ir *ingressroutev1.IngressRoute, prefixMatch string, visited map[meta]bool, host string, service func(m meta, port intstr.IntOrString) *Service, vhost func(host string, port int) *VirtualHost) {
+func (d *DAG) processIngressRoute(ir *ingressroutev1.IngressRoute, prefixMatch string, visited *ingressrouteStack, host string, service func(m meta, port intstr.IntOrString) *Service, vhost func(host string, port int) *VirtualHost, orphaned map[meta]bool) []ingressrouteStatus {
+	visited.push(ir)
+	defer func() {
+		visited.pop()
+	}()
+
+	// validate load balancing strategies are consistent across all references to the same service
+	lbstrat := make(map[string]string)
+	for _, route := range ir.Spec.Routes {
+		for _, svc := range route.Services {
+			s, ok := lbstrat[svc.Name]
+			if ok && svc.Strategy != s {
+				return []ingressrouteStatus{{object: ir, status: "invalid", msg: fmt.Sprintf("different load balancing strategies were specified for service %q", svc.Name)}}
```

Comment:
nit: Should we include the namespace in the error so it's clear where the issue is?